facebookresearch/ParlAI
# Report correct agent-reported metrics during evaluation when multi-tasking

**Patch description**
Currently during multitasking, agent reported metrics aren't reported correctly. This changes makes sure that the metrics are reported correctly during evaluation by creating multiple worlds and iterating through them sequentially, collecting separate metrics for each world and then finally aggregating them. 

Note that in this case, if we specify `validation_max_exes`, it splits the number of examples equally across tasks. In the future, this could change to be different proportions.

**Testing steps**
Train and evaluate models, check that metrics are reported correctly. [Edit: should probably write unit tests but will wait until feedback from initial review...]
